components/net/image_cache.rs

```diff
@@ -109,7 +109,8 @@ fn is_image_opaque(format: webrender_api::ImageFormat, bytes: &[u8]) -> bool {
 fn premultiply(data: &mut [u8]) {
     let length = data.len();
 
-    for i in Iterator::step_by(0..length, 4) {
+    let mut i = 0;
+    while i < length {
```

Comment:
What about using [`<[_]>::chunks`](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks)?

Reply:
The last item of `chunks` can be a smaller slice, whereas this loop will stop before.